spec/features/channel_filter_on_results_page_spec.rb

```diff
@@ -0,0 +1,178 @@
+      and_only_in_person_firms_are_displayed
+    end
+  end
+
+  scenario 'Consumer changes advice method to in person without providing postcode' do
+    with_elastic_search! do
+      given_some_firms_were_indexed
+      and_i_have_performed_a_remote_search_from_the_landing_page
+      when_i_select_face_to_face_advice_without_providing_postcode
+      and_i_submit_the_sidebar_search_form
+      then_i_am_on_the_landing_page
+      and_a_validation_error_message_is_displayed
+    end
+  end
+
+  scenario 'Consumer changes advice method to remote without selecting remote advice type' do
+    with_elastic_search! do
+      given_some_firms_were_indexed
+      and_i_have_performed_an_in_person_search_from_the_landing_page
+      when_i_select_remote_advice_without_providing_type
+      and_i_submit_the_sidebar_search_form
+      then_i_am_on_the_landing_page
+      and_a_validation_error_message_is_displayed
+    end
+  end
+
+
+  def given_some_firms_were_indexed
+    with_fresh_index! do
+      @phone_firm     = create(:firm, registered_name: 'The Willers Ltd',    in_person_advice_methods: [], other_advice_methods: [ phone_advice ])
+      @online_firm    = create(:firm, registered_name: 'The Equiters Ltd',   in_person_advice_methods: [], other_advice_methods: [ online_advice ])
+      @in_person_firm = create(:firm, registered_name: 'Estate Traders Ltd', other_advice_methods: [])
+
+      @reading   = create(:adviser, postcode: 'RG2 8EE', firm: @phone_firm,       latitude: 51.428473, longitude: -0.943616)
+      @leicester = create(:adviser, postcode: 'LE1 6SL', firm: @online_firm,      latitude: 52.633013, longitude: -1.131257)
+      @glasgow   = create(:adviser, postcode: 'G1 5QT',  firm: @in_person_firm, latitude: 55.856191, longitude: -4.247082)
+    end
+  end
+
+  def and_i_am_on_the_rad_landing_page
+    landing_page.load
+  end
+
+  def and_i_have_performed_an_in_person_search_from_the_landing_page
+    landing_page.load
+    landing_page.in_person.tap do |section|
+      section.postcode.set('G1 5QT')
+      section.equity_release.set(true)
+      section.search.click
+    end
+  end
+
+  def and_i_have_performed_a_remote_search_from_the_landing_page
+    landing_page.load
+    landing_page.remote.tap do |section|
+      section.online.set(true)
+      section.search.click
+    end
+  end
+
+  def when_i_select_a_remote_advice_method
+    landing_page.load
+    landing_page.in_person.tap do |section|
+      section.online.set(true)
+      section.search.click
+    end
+  end
+
+  def when_i_select_remote_advice
+    results_page.criteria.remote_advice.set(true)
+    results_page.criteria.online.set(true)
+  end
+
+  def when_i_select_face_to_face_advice_without_providing_postcode
+    results_page.criteria.in_person_advice.set(true)
+  end
+
+  def when_i_select_remote_advice_without_providing_type
+    results_page.criteria.remote_advice.set(true)
+  end
+
+  def and_i_submit_the_in_person__search_form
```

Comment:
This has a double underscore, which is weird, but it seems that the step isn't being used anywhere?